code/controllers/configuration/entries/jobs.dm

```diff
@@ -0,0 +1,184 @@
+				var/config_value = job_config[job_key][config_datum_key]
+				config_datum.set_current_value(occupation, config_value)
+
+	else // legacy mode, so just run the old parser.
+		var/jobsfile = file("[global.config.directory]/jobs.txt")
+		if(!fexists(jobsfile)) // sanity with a trace
+			stack_trace("Despite SSconfig setting SSjob.legacy_mode to TRUE, jobs.txt was not found in the config directory! Something has gone terribly wrong!")
+			return
+		var/jobstext = file2text(jobsfile)
+		for(var/datum/job/occupation as anything in joinable_occupations)
+			var/regex/parser = new("[occupation.title]=(-1|\\d+),(-1|\\d+)")
+			parser.Find(jobstext)
+			occupation.total_positions = text2num(parser.group[1])
+			occupation.spawn_positions = text2num(parser.group[2])
+
+/// Called from an admin debug verb that generates the jobconfig.toml file and then allows the end user to download it to their machine.
+/// Returns TRUE if a file is successfully generated, FALSE otherwise.
+/datum/controller/subsystem/job/proc/generate_config(mob/user)
+	var/toml_file = "[global.config.directory]/jobconfig.toml"
+	var/jobstext = "[global.config.directory]/jobs.txt"
+	var/list/file_data = list()
+	config_documentation = initial(config_documentation) // Reset to default juuuuust in case.
+
+	if(fexists(file(toml_file)))
+		to_chat(src, span_notice("Generating new jobconfig.toml, pulling from the old config settings."))
+		if(!regenerate_job_config(user))
+			return FALSE
+		return TRUE
+
+	if(fexists(file(jobstext))) // Generate the new TOML format, migrating from the text format.
+		to_chat(user, span_notice("Found jobs.txt in config directory! Generating jobconfig.toml from it."))
+		jobstext = file2text(file(jobstext)) // walter i'm dying (get the file from the string, then parse it into a larger text string)
+		config_documentation += "\n\n## This TOML was migrated from jobs.txt. All variables are COMMENTED and will not load by default! Please verify to ensure that they are correct, and uncomment the key as you want, comparing it to the old config.\n\n" // small warning
+		for(var/datum/job/occupation as anything in joinable_occupations)
+			var/job_key = occupation.config_tag
+			var/regex/parser = new("[occupation.title]=(-1|\\d+),(-1|\\d+)") // TXT system used the occupation's name, we convert it to the new config_key system here.
+			parser.Find(jobstext)
+
+			var/default_positions = text2num(parser.group[1])
+			var/starting_positions = text2num(parser.group[2])
+
+			// Playtime Requirements and Required Account Age are new and we want to see it migrated, so we will just pull codebase defaults for them.
+			// Remember, every time we write the TOML from scratch, we want to have it commented out by default to ensure that the server operator is knows that they codebase defaults when they remove the comment.
+			var/list/working_list = list(
+				"# [JOB_CONFIG_TOTAL_POSITIONS]" = default_positions,
+				"# [JOB_CONFIG_SPAWN_POSITIONS]" = starting_positions,
```

Comment:
is this the comment in string meme

Reply:
yes. this PR would be absolute hell if both were to be implemented at the same time without any assurance of stability